metpy/calc/thermo.py

```diff
@@ -1492,7 +1491,7 @@ def _find_append_zero_crossings(x, y):
     y = y[sort_idx]
 
     # Remove duplicate data points if there are any
-    keep_idx = np.ediff1d(x, to_end=[1]) > 0
+    keep_idx = np.ediff1d(x, to_end=[1]) > 10**(-6)
```

Comment:
So the normal way to write this is `1e-6`. An idea that just occurred to me, though is:
```suggestion
    keep_idx = ~np.isclose(np.ediff1d(x, to_end=[1]),  0)
```
That's not identical, so I'm trying to think of cases where keeping points where the diff is <0 is a bug...

Reply:
Trying to understand why what we had before (using the threshold) is not the same/better than adding `np.isclose`? Is that more pythonic or faster or just easier to understand?

Reply:
My thinking was that `isclose` better expresses our intent. It might also handle more corner cases in the approach. Meh. Just change to `1e-6` and call it good.

Reply:
Haha fair enough - will do.